### PR TITLE
feat(curriculum): add tests and solution

### PR DESCRIPTION
**Pull request by gikf**
- Adds more tests and adjusts description to conform that.
- Adds solution.
- Partially related to #40896.
- Tested on local fork.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-79-passcode-derivation.md [hunk 1]
```diff
@@ -8,28 +8,54 @@ dashedName: problem-79-passcode-derivation
...
 Given that the three characters are always asked for in order, analyze the array so as to determine the shortest possible secret passcode of unknown length.
 
 # --hints--
 
-`passcodeDerivation(keylog)` should return a number.
+`passcodeDerivation(keylog1)` should return a number.
 
 ```js
-assert(typeof passcodeDerivation(keylog) === 'number');
+assert(typeof passcodeDerivation(_keylog1) === 'number');
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-79-passcode-derivation.md:22]: Not sure if you meant to set an underscore here.

Comment (gikf) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-79-passcode-derivation.md:22]: It is intended, just to use test cases defined in the background, and avoid possible conflicts with variable having the same name as in camper seed code.

**Review by Sembauke — APPROVED** [review]
LGTM 👍

**Comment by moT01** [comment]
I added this solution to the forum guide article

**Pull request merged**